security/pkg/pki/ra/k8s_ra.go

```diff
@@ -116,24 +116,17 @@ func (r *KubernetesRA) SignWithCertChain(csrPEM []byte, certOpts ca.CertOpts) ([
 		if err != nil {
 			return nil, fmt.Errorf("failed to find root cert from mesh config (%v)", err.Error())
 		}
-		if rootCertFromCertChain != nil && rootCertFromMeshConfig != nil {
-			if !bytes.Equal(rootCertFromCertChain, rootCertFromMeshConfig) {
-				return nil, fmt.Errorf("root cert from signed cert-chain" +
-					" is conflicting with the one specified in mesh config")
-			}
-		}
 		if rootCertFromMeshConfig != nil {
 			rootCert = rootCertFromMeshConfig
-		}
-
-		if rootCertFromCertChain != nil {
+		} else if rootCertFromCertChain != nil {
```

Comment:
this changes the behavior. before cert chain > mesh config, now its the inverse?

Reply:
We now will verify cert-chain against the root cert in meshconfig not simply comare bytes. - in some cases, the root cert in cert-chain might be only an intermediate ca.